docs/schema/index-management/index-lifecycle.md

```diff
@@ -26,28 +26,72 @@ mgmt.commit();
 
-## Actions (SchemaAction)
-The following actions can be performed on an index to change its state via `mgmt.updateIndex()`:
+**ENABLED**
+
+:   The index is enabled and can be used to answer queries.
+    Furthermore, this is the only state in which insertions and deletions in the graph are forwarded to the index.
 
--   **REGISTER_INDEX** Registers the index with all instances in the graph cluster. After an index is installed, it must be registered with all graph instances
+**DISABLED**
 
--   **REINDEX** Re-builds the index from the graph
+:   The index is disabled and will not be used to answer queries.
+It also does no longer receive graph updates.
+Therefore, it is recommended to use `REINDEX` to re-enable the index.
 
--   **ENABLE_INDEX** Enables the index so that it can be used by the query processing engine. An index must be registered before it can be enabled.
+**DISCARDED**
 
--   **DISABLE_INDEX** Disables the index in the graph so that it is no longer used.
+:   The index has been decommissioned and its contents have been deleted.
+    The only remaining artifact is the schema vertex representing the index.
+
+## Actions (SchemaAction)
+The following actions can be performed on an index to change its state via `mgmt.updateIndex()`:
 
--   **REMOVE_INDEX** Removes the index from the graph (optional operation). Only on composite index.
+**REGISTER_INDEX**
+:   Registers the index with all instances in the graph cluster.
+After an index is installed, it must be registered with all graph instances.
+
+**REINDEX**
+:   Re-builds the index from the ground up using the data stored in the graph.
+Depending on the size of the graph, this action can take a long time to complete.
+Reindexing is possible in all stable states and automatically enables the index once finished.
+
+**ENABLE_INDEX**
+:   Enables the index so that it can be used by the query processing engine.
+An index must be registered before it can be enabled.
+If enabling the index manually instead of performing a reindex, be aware that past modifications of the graph are not represented in the index.
+Enabling a previously disabled index without `REINDEX` may cause index queries to return ghost vertices, which have been deleted from the graph while the index was disabled.
+
+**DISABLE_INDEX**
+:   Disables the index temporarily so that it is no longer used to answer queries.
+The index also stops receiving updates in the graph.
+
+**DISCARD_INDEX**
+:   Removes all data from the index, preparing it for removal.
+    This operation is supported for all composite indices and for some mixed index backends.
+    For more information on index removal see [Removal](index-removal.md).
+
+**MARK_DISCARDED**
+:   All indices which can not be discarded using **DISCARD_INDEX** have to be discarded manually in the backend.
+    As this is not recognized by JanusGraph automatically, the state can be set to **DISCARDED** manually via this action.
+
+**DROP_INDEX**
```

Comment:
Wouldn't it be possible to merge the drop action into the two discard actions and then remove the `DISCARDED` state completely? Or what's the purpose of the `DISCARDED` state except of waiting for the user to execute this action?

Reply:
I already tried that, but there is currently no mechanism in place to properly wait for the long running index removal and schedule the deletion of the internal vertex (aka `DROP_INDEX`) afterwards.

Reply:
Thanks for the explanation. Then I'm of course good with this. We can still try to improve this at some point in the future, but then it's out of scope for this PR.